core/src/main/java/de/evoila/cf/broker/controller/BackupController.java

```diff
@@ -4,22 +4,27 @@
 import de.evoila.cf.broker.service.BackupService;
 import de.evoila.cf.model.BackupRequest;
 import de.evoila.cf.model.RestoreRequest;
-import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
 import org.springframework.data.domain.Pageable;
 import org.springframework.data.web.PageableDefault;
 import org.springframework.http.ResponseEntity;
+import org.springframework.util.Assert;
 import org.springframework.web.bind.annotation.*;
 
 import java.util.HashMap;
 
 /** @author Yannic Remmet. */
 @RestController
 @RequestMapping(value = "/v2/manage/backup")
+@ConditionalOnBean(name="backupService") // Needs to be the Bean Name in this case otherwise Spring tries to create the controller
 class BackupController extends BaseController {
```

Comment:
Why don’t you use backupServiceCondition here like in the BackupServiceImpl below?